source/gameengine/Rasterizer/RAS_DisplayArray.cpp

```diff
@@ -48,31 +48,24 @@ int RAS_DisplayArray::GetOpenGLPrimitiveType() const
 
 void RAS_DisplayArray::UpdateFrom(RAS_DisplayArray *other, int flag)
 {
-	if (flag & RAS_MeshObject::TANGENT_MODIFIED) {
+	if (flag & (RAS_MeshObject::POSITION_MODIFIED | RAS_MeshObject::NORMAL_MODIFIED | RAS_MeshObject::TANGENT_MODIFIED | RAS_MeshObject::UVS_MODIFIED | RAS_MeshObject::COLORS_MODIFIED)) {
 		for (unsigned int i = 0; i < other->m_vertex.size(); ++i) {
-			m_vertex[i].SetTangent(MT_Vector4(other->m_vertex[i].getTangent()));
-		}
-	}
-	if (flag & RAS_MeshObject::UVS_MODIFIED) {
-		for (unsigned int i = 0; i < other->m_vertex.size(); ++i) {
-			for (unsigned int uv = 0; uv < 8; ++uv) {
-				m_vertex[i].SetUV(uv, MT_Vector2(other->m_vertex[i].getUV(uv)));
+			switch (flag)
+			{
+			case RAS_MeshObject::POSITION_MODIFIED:
+				m_vertex[i].SetXYZ(MT_Vector3(other->m_vertex[i].getXYZ()));
```

Comment:
So this will improve moving vertex in mesh speed / gpu update as well as
armature skin and soft body?


On Wed, Jun 22, 2016 at 3:33 AM, panzergame <notifications@github.com>
wrote:

> In source/gameengine/Rasterizer/RAS_DisplayArray.cpp
> <https://github.com/UPBGE/blender/pull/140#discussion_r68030511>:
>
> >  		for (unsigned int i = 0; i < other->m_vertex.size(); ++i) {
> > -			m_vertex[i].SetTangent(MT_Vector4(other->m_vertex[i].getTangent()));
> > -		}
> > -	}
> > -	if (flag & RAS_MeshObject::UVS_MODIFIED) {
> > -		for (unsigned int i = 0; i < other->m_vertex.size(); ++i) {
> > -			for (unsigned int uv = 0; uv < 8; ++uv) {
> > -				m_vertex[i].SetUV(uv, MT_Vector2(other->m_vertex[i].getUV(uv)));
> > +			switch (flag)
> > +			{
> > +			case RAS_MeshObject::POSITION_MODIFIED:
> > +				m_vertex[i].SetXYZ(MT_Vector3(other->m_vertex[i].getXYZ()));
>
> you should use break to avoid doing all the code after. For example a
> vertex modified in position will be updated in position, normal, tangent
> and uv here without break.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UPBGE/blender/pull/140/files/456021d918b0f34397fa8f8c93e3f4e7520559f1#r68030511>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG25WWKecbwRagqNG62V0Pj3xuwzS8rfks5qOQ97gaJpZM4I673F>
> .
>